src/actions/blockActions.ts

```diff
@@ -111,21 +111,43 @@ export const resetBlockState =
     })
   }
 
-export function fetchBlock(indexOrHash: string | number = 1) {
+export function fetchBlock(index = 1) {
   return async (
     dispatch: ThunkDispatch<State, void, Action>,
-    getState: () => { block: State },
+    getState: () => { block: State; network: NetworkState },
   ): Promise<void> => {
-    if (shouldFetchBlock(getState(), indexOrHash)) {
-      dispatch(requestBlock(indexOrHash))
+    if (shouldFetchBlock(getState(), index)) {
+      dispatch(requestBlock(index))
       try {
-        const response = await fetch(
-          `${GENERATE_BASE_URL()}/block/${indexOrHash}`,
-        )
-        const json = await response.json()
-        dispatch(requestBlockSuccess(json))
-      } catch (e: any) {
-        dispatch(requestBlockError(indexOrHash, e))
+        const network = getState().network.network
+        const response = await NeoRest.block(index, network)
+
+        const block = {
+          nextconsensus: response.nextconsensus,
+          oversize: 0,
+          previousblockhash: response.previousBlockHash,
+          index: response.index,
+          version: response.version,
+          nonce: response.nonce,
+          size: response.size,
+          blocktime: response.blocktime,
+          merkleroot: response.merkleroot,
+          time: Number(response.time),
+          hash: response.hash,
+          jsonsize: response.jsonsize,
+          tx: response.tx.map(
+            t =>
+              ({
+                size: t.size,
+                time: Number(t.time),
+                txid: t.hash,
+                hash: t.hash,
+              } as BlockTransaction),
+          ),
+        } as DetailedBlock
```

Comment:
nitpick: I think this could be less verbose with something like:
```javascript
const { network } = getState().network;
const { nextconsensus, previousBlockHash, tx, ...rest } = await NeoRest.block(index, network);

const block = {
  nextconsensus,
  previousblockhash: previousBlockHash,
  tx: tx.map(({ size, time, hash }) => ({ size, time: Number(time), txid: hash, hash })),
  ...rest,
} as DetailedBlock;
```